openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +116,267 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+        self.assertCountEqual(extract_dates_from_course(course), expected_dates)
+
+        # With due_num_weeks and a problem. The problem should inherit the due date.
+        problem = ItemFactory.create(category='problem', parent=vertical)
+        expected_dates.append((problem.location, {'due': timedelta(days=21)}))
+        course = modulestore().get_item(course.location)
+        self.assertCountEqual(extract_dates_from_course(course), expected_dates)
+
+    @override_waffle_flag(CUSTOM_PLS, active=True)
+    def test_extract_dates_from_course_custom_and_default_pls(self):
+        """
+        extract_dates_from_course should have the right set due dates for
+        subsections with custom PLS and subsections with the default PLS logic
+        """
+        # due_num_weeks in one of the subsections. Only one of them should have a set due date.
+        # The other subsections do not have due dates because they are not graded
+        # and default PLS do not assign due dates to non graded assignments.
+        course = self.courses[0]
+        chapter = course.get_children()[0]
+        sequential = chapter.get_children()[0]
+        ItemFactory.create(category='sequential', parent=chapter)
+        ItemFactory.create(category='sequential', parent=chapter)
+        expected_dates = [
+            (course.location, {}),
+            (chapter.location, timedelta(days=28)),
+            (sequential.location, {'due': timedelta(days=21)})
+        ]
+        course = modulestore().get_item(course.location)
+        self.assertCountEqual(extract_dates_from_course(course), expected_dates)
+
+        # course 3: a section with a subsection that has due_num_weeks and a subsection without due_num_weeks that has graded content.
+        # Current PLS should apply for the subsection without due_num_weeks that has graded content.
+        course = self.courses[2]
+        chapter1 = course.get_children()[0]
+        sequential1 = chapter1.get_children()[0]
+        vertical1 = sequential1.get_children()[0]
+        problem1 = vertical1.get_children()[0]
+
+        chapter2 = course.get_children()[1]
+        sequential2 = chapter2.get_children()[0]
+        vertical2 = sequential2.get_children()[0]
+        problem2 = vertical2.get_children()[0]
+
+        expected_dates = [
+            (course.location, {}),
+            (chapter1.location, timedelta(days=14)),
+            (sequential1.location, {'due': timedelta(days=14)}),
+            (vertical1.location, {'due': timedelta(days=14)}),
+            (problem1.location, {'due': timedelta(days=14)}),
+            (chapter2.location, timedelta(days=28)),
+            (sequential2.location, {'due': timedelta(days=28)}),
+            (vertical2.location, {'due': timedelta(days=28)}),
+            (problem2.location, {'due': timedelta(days=28)})
+        ]
+        course = modulestore().get_item(course.location)
+        self.assertCountEqual(extract_dates_from_course(course), expected_dates)
```

Comment:
So this isn’t a fun test because even though you explicitly set the due num weeks, it looks the exact same as if you used the default logic. Change the value so it’s more clear in the test that something different is happening here